### PR TITLE
Add extra data augmentation techniques.

### LINKED ISSUE #55: Add data augmentation techniques
- [ ] Add random patching
- [ ] Add patching based on IoU
- [ ] Add random resize within limits
- [ ] Add photo-metric distortions

### PR DESCRIPTION
**Pull request by IanTayler**
Also fixes `random_patch` and adds basic tests for the data augmentation techniques introduced.

Resolves #55

### CODE REVIEW

**Comment by IanTayler** [comment]
Includes the changes requested in #67

File: luminoth/utils/image_test.py [hunk 1]
```diff
@@ -261,6 +319,126 @@ def testFlipBboxesDiffDtype(self):
             flipped_boxes_float, flipped_boxes_int
         )
 
+    def testRandomPatchImageBboxes(self):
```
Comment (vierja) [inline luminoth/utils/image_test.py]: Add a test where bbox are dropped but some still are returned.
And one to test the `update_condition`.

File: luminoth/utils/image_test.py [hunk 2]
```diff
@@ -261,6 +319,126 @@ def testFlipBboxesDiffDtype(self):
...
+            ],
+            axis=1
+        )
+
+        ret_image, ret_bboxes = self._random_distort(
+            image, config, bboxes_w_label
+        )
+        # Assertions
+        self.assertEqual(im_shape, ret_image.shape)
+        self.assertAllClose(
```
Comment (vierja) [inline luminoth/utils/image_test.py]: Why not equal?

File: luminoth/utils/image_test.py [hunk 3]
```diff
@@ -261,6 +319,126 @@ def testFlipBboxesDiffDtype(self):
...
+            bboxes, ret_bboxes[:, :4], atol=self._equality_delta
+        )
+
```
Comment (vierja) [inline luminoth/utils/image_test.py]: Add assert with small distortions in which the resulting image is allClose within a rtol of ~5% of the original image.

File: luminoth/utils/image.py [hunk 4]
```diff
@@ -296,13 +321,155 @@ def random_patch(image, bboxes=None, debug=False):
...
+    return_dict['bboxes'] = tf.cond(
+        update_condition,
+        lambda: new_bboxes,
+        lambda: bboxes
+    )
     return_dict['bboxes'] = new_bboxes
     return return_dict
+
+
+def random_resize(image, bboxes=None, debug=False, min_size=400,
```
[the remark this answers is not in the document]
Comment (IanTayler) [reply luminoth/utils/image.py]: Yes, that's better.

**Pull request merged**